frontend/catalyst/utils/jax_cpu_lapack_kernels/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+project(catalyst_jax_cpu_lapack_kernels)
+
+# Create a static library for LAPACK kernels to be included into the
+# custom_calls library in the parent scope.
+
+add_library(catalyst_jax_cpu_lapack_kernels STATIC
+    ${PROJECT_SOURCE_DIR}/lapack_kernels.cpp
+    ${PROJECT_SOURCE_DIR}/lapack_kernels_using_lapack.cpp
+)
+
+# Ensure fPIC is set for static builds
+set_property(TARGET catalyst_jax_cpu_lapack_kernels PROPERTY POSITION_INDEPENDENT_CODE ON)
+
+# Headers are tracked into the target from the current project scope 
+target_include_directories(catalyst_jax_cpu_lapack_kernels PUBLIC
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+  $<INSTALL_INTERFACE:include> 
+)
+
+# If the option is set, allow explicit linking of the CMake-found CBLAS/LAPACKE
+# library. See https://cmake.org/cmake/help/latest/module/FindBLAS.html for info.
+option(CATALYST_LINK_BLAS "Link the CMake-defined BLAS and LAPACK libraries as part of the extension module" OFF)
+
+
+if(CATALYST_LINK_BLAS)
+    find_package( BLAS REQUIRED )
+    find_package( LAPACK REQUIRED)
+    target_link_libraries(catalyst_jax_cpu_lapack_kernels PUBLIC BLAS::BLAS)
+
+    # Ensure libraries are explicitly linked.
+    target_link_options(catalyst_jax_cpu_lapack_kernels PUBLIC
+        $<$<COMPILE_LANGUAGE:CXX>:-Wl,--no-as-needed>
+    )
+endif()
```

Comment:
I don't think we had this before. Given our saga with lapack, do you mind expanding on the reasoning here? 😅

Reply:
This is just an opt-in control option to ensure any user can build the library with the system's provided BLAS/LAPACK libraries. This is preferential for in-place builds on systems we wouldn't control and allows the user to rely on any of the ones listed in https://cmake.org/cmake/help/latest/module/FindBLAS.html#blas-lapack-vendors

Since it is an option, this doesn't do anything unless set. If you'd prefer I can remove it, but it would be something I'd use (or want if not available).

Reply:
Maybe better in a separate PR --- I'll pull it out and create a separate branch.

Reply:
Got it, if optional it's all good then 👍

Reply:
This block looks for lapack and blas, but only links against blas, is that intended?